EIPS/eip-beacon_chain_push_withdrawals.md

```diff
@@ -0,0 +1,121 @@
+The architecture is "push"-based, rather than "pull"-based, where withdrawals are required to be processed in the execution block as soon as they are dequeued from the beacon chain.
+
+This approach is more involved than "pull"-based alternatives (e.g. [EIP-4788](./eip-4788.md) + user-space withdrawal contract) with respect to the core protocol (by providing a new transaction type with special semantics) but does provide tighter integration of a critical feature into the protocol itself.
+
+## Specification
+
+| constants                     | value                                          | units
+|---                            |---                                             |---
+| `FORK_TIMESTAMP`              | TBD                                            |
+| `WITHDRAWAL_TX_TYPE`          | `0x3`                                          | byte
+
+Beginning with the execution timestamp `FORK_TIMESTAMP`, execution clients **MUST** introduce the following extensions to transaction processing and block validation:
+
+### New transaction type
+
+Define a new [EIP-2718](./eip-2718.md) transaction type with `TransactionType` `WITHDRAWAL_TX_TYPE`.
+
+The `TransactionPayload` is an SSZ-encoded container given by the following schema:
+
+```python
+class WithdrawalTransaction(Container):
+    address: ExecutionAddress
+    amount: Gwei
+```
+
+where `ExecutionAddress` is an alias for a `Bytes20` SSZ type and `Gwei` is an alias for a `uint64` SSZ type.
+Refer to the [SSZ specs](https://github.com/ethereum/consensus-specs/blob/master/ssz/simple-serialize.md) for further details on layout and encoding.
+
+### Block validity
+
+If a block contains *any* transactions with `WITHDRAWAL_TX_TYPE` type, they **MUST** come before **ALL** other transactions in the block.
+
+If the execution client receives a block where this is not the case, it **MUST** consider the block invalid.
+
+### Transaction processing
+
+When processing a transaction with `WITHDRAWAL_TX_TYPE` type, the implementation should make a balance transfer for `amount` Gwei to the `address` specified by the `WithdrawalTransaction`.
+There is no source for this balance transfer, much like the coinbase transfer.
+
+This balance transfer is unconditional and **MUST** not fail.
+
+This transaction type has no associated gas costs.
+
+TODO: add logs?
+
+## Rationale
+
+### Push vs pull
+
+cheaper for validators
+
+happens automatically
+
+### Why a new transaction type? And why no gas?
+
+Special semantics
+
+Firewall off generic EVM excution from this type of processing to simplify testing, security review.
+
+### Why no gas costs for new transaction type?
+
+The maximum number of this transaction type that can reach the execution layer at a given time is bounded (enforced by the consensus layer) and this limit is kept small so that
+any gas required is negligible in the context of the broader block gas limit.
```

Comment:
```suggestion
any execution layer operational costs is negligible in the context of the broader block execution.
```
For clarity, we should be talking about execution layer operational costs here, since that is what matters (gas is just a proxy for that, and it isn't even being used in this case).